implementation.md

```diff
@@ -1554,3 +1554,268 @@ In the `Video` and `Audio` objects, sellers must only include one form of floor
+
+Items in the EID array with <code>mm</code> values of 1 or 2 are generally optional/informational, but some buyers may find it useful for sellers to provide them.
+
+### 7.12.2 - Roles Overview
+- <code>inserter</code> is the party that’s putting the ID into the bid request - Typically a Publisher or an SSP. In other words, the party that made the decision to put the id in the bidstream. In the case of header bidders, the inserter is the Publisher.<br></br>
+- <code>source</code>is the party that defined/created an ID. In the case of universal or alt-IDs, it’s the domain of the party who defined/created the ID itself. In the case of cookie IDs, it’s the domain of the party who the cookie belongs to.<br>
+<b>EXAMPLES FOR PUBLIC COMMENT ONLY</b> adsrvr.org, id5.com, liveramp.com, and others.<br></br>
+- <code>matcher</code> is the party that created the match included in the ID array, this could be a device graph vendor, or the publisher themselves.
+
+### 7.12.3 - Singaling both Agent Type and Match Method
+Values in the <code>atype</code> attribute should be used to describe the level of the ID in the bidstream and can apply to multiple match methods. 
+
+Different iterations of possible combinations are listed in the following table. Additional json examples for how to use <code>atype</code> alongside match method (<code>mm</code>) signaling can be found in the next section.
+
+<table>
+  <tr>
+    <td><strong>atype Value</strong></td>
+    <td><strong>atype Definition</strong></td>
+    <td><strong>Possible Match Methods</strong></td>
+  </tr>
+  <tr>
+    <td>1</td>
+    <td>An ID which is tied to a specific web browser or device (cookie-based, probabilistic, or other).</td>
+    <td>Any match method that pertains to one and only one browser and/or device, regardless of the method used to do the match. Any single mm value of 1, 2, 5, 6 could be applicable</td>  
+  </tr>
+  <tr>
+    <td>2</td>
+    <td>In-app impressions, which will typically contain a type of device ID (or rather, the privacy-compliant versions of device IDs).</td>
+    <td>Match method that pertains to a single application on a single device, regardless of how the match was done. Applicable mm values could be either 1, 5 or 6</td>  
+  </tr>
+  <tr>
+    <td>3</td>
+    <td>A person-based ID, i.e., that is the same across devices.</td>
+    <td>Match method that pertains to multiple devices, regardless of how the match was done. Applicable mm values could be either 3, 4</td> 
+  </tr>
+</table>
+
+### 7.12.4 - json Examples
+
+#### 7.12.4.1 User and Device IDs
+
+##### Example 1. DSP cookie derived from real-time cookie sync, subsequently observable
+```
+{
+    "user": {
+        "buyeruid": "fac13741-0648-436a-88cf-aceafdf45c9a",
+        "eids": [
```

Comment:
This usage of `eids` is weird. What if there are other EIDs in the array?

Reply:
Yeah, this should not be here. atype=1 is inferred from the existence of buyeruid.

Reply:
i get very nervous with any kind of inference when it comes to this stuff. also i wanted to include atype in all the examples, even where they may be redundant to give the community a visual for how to use them in conjunction with each other. not a hill i'll die on, but do we really feel strongly we should nuke it?

Reply:
By definition, I think we can infer everything on buyeruid. Example description should be updated noting this. We can fix after public comment IMO

Reply:
Agreed. The empty EIDs array makes no sense to me. I think simply remove. No need for atype.